node_modules/gh-users/lib/api.js

```diff
@@ -324,6 +335,243 @@ var updateAdminStatus = module.exports.updateAdminStatus = function(ctx, userId,
+                        log().error({'err': err, 'app': appId}, 'Failed to parse CSV file when importing users');
+                        return callback({'code': 500, 'msg': err.message});
+                    }
+
+                    // Validate each record before attempting to create users. Although this makes
+                    // the request a bit slower, it prevents us from ending up with only half of the
+                    // CSV file's users
+                    _.each(users, function(user) {
+                        validator.check(user[0], {'code': 400, 'msg': 'Missing display name for: ' + user[1]}).notEmpty();
+                        if (authenticationStrategy === 'local') {
+                            validator.check(user[1], {'code': 400, 'msg': 'A valid email address must be provided for: ' + user[0]}).isEmail();
+                            validator.check(user[2], {'code': 400, 'msg': 'A valid password must be provided for: ' + user[0]}).notEmpty();
+                            validator.check(user[2], {'code': 400, 'msg': 'A valid password must be provided for: ' + user[0]}).len(6);
+                        } else {
+                            validator.check(user[2], {'code': 400, 'msg': 'A valid shibboleth id must be provided for: ' + user[0]}).notEmpty();
+                        }
+                    });
+
+                    if (validator.hasErrors()) {
+                        return callback(validator.getFirstError());
+                    }
+
+                    // Indicate the request is valid and we will start actually importing users
+                    UsersAPI.emit('preImportUsers');
+
+                    // If the CSV parse was successful, we call the callback to prevent the request from timing out
+                    // whilst the users are being loaded
+                    callback();
+
+                    // Import the users
+                    log().info({
+                        'app': appId,
+                        'authenticationStrategy': authenticationStrategy
+                    }, 'Starting user import from CSV');
+                    _importUsers(appId, authenticationStrategy, forceProfileUpdate, users, function(err) {
+                        if (!err) {
+                            log().info({
+                                'app': appId,
+                                'authenticationStrategy': authenticationStrategy
+                            }, 'All users have been imported');
+                        }
+
+                        // Indicate that all the users from the CSV file have been processed
+                        UsersAPI.emit('postImportUsers', err);
+                    });
+                });
+            });
+
+            // The CSV module works with streams, so get a readable stream to the uploaded CSV file
+            // and pipe it to the CSV parser
+            var input = fs.createReadStream(file.file);
+            input.pipe(parser);
```

Comment:
I think `fs.createReadStream(file.file).pipe(parser);` is nicer for this, if you agree